aten/src/ATen/native/Distributions.cpp

```diff
@@ -302,12 +302,27 @@ Tensor multinomial_cpu(const Tensor& self, int64_t n_sample, bool with_replaceme
   return result;
 }
 
+/* The largest consecutive integer representable in float32 (2^24) */
+#define FLOAT32_MAX_CONSECUTIVE_INT 16777216.0f
+
 Tensor& multinomial_out_cpu(Tensor& result, const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
-  TORCH_CHECK(at::isFloatingType(self.scalar_type()), "multinomial only supports floating-point dtypes for input, got: ", self.scalar_type());
-  TORCH_CHECK(result.scalar_type() == ScalarType::Long, "multinomial expects Long tensor out, got: ", result.scalar_type());
+  auto device1 = result.type().device_type();
+  auto device2 = self.type().device_type();
+  TORCH_CHECK(device1 == device2, "output and input must have the same device type. output: ", device1, " input: ", device2);
+  TORCH_CHECK(!result.is_cuda() || result.get_device() == self.get_device(), "device of output (", result.get_device(),
```

Comment:
Can't this (and the above line) just be, `result.device() == self.device()`?

Reply:
Fixed